### ISSUE TITLE
confusing/incomplete ConvTranspose2d DOC

### ISSUE THREAD

**Issue by dgerin** [pytorch/pytorch#56873]
Hi,

My initial objective is simple; I would like to computes ConvTranspose2d with Conv2d layer (don't ask why please :D), and I succeed, but in an unexpected way.

The unexpected think in the code below is the use of transposeConv2dKernel()
function to "pre transpose" the conv2d W. 

I really though that only Input Striding + Zero padding see **inputStridedPadded**  below would functionally match the ConvTransPose2d but this is still not functionally equivalent. I needed to transpose the conv2d Kernel W with transposeConv2dKernel()
to perform a 2d rotation, to exactly match.

This show that the course of conv2d is not equivalent to the course of ConvTransPose2d so i though it was. 

**I don't see the course of layers conv2d / transposeConv2dKernel in the doc, so could you give me the exact spec  please ?**

See code below.
```
    k = 3
    len = 4
    stride = 3
    h, w = len, len
    n, m = 2, 2
    N = 1
    W = np.random.normal(0, 1, size=(n, m, k, k)).astype(np.float32)

    def rot2d180(a):
            a= np.rot90(a)
            a= np.rot90(a)
            return a

    def transposeConv2dKernel(a):
        a = np.moveaxis(a, [0,1, 2, 3], [1, 0, 2,3])
        for i in range(a.shape[0]):
            for j in range(a.shape[1]):
                a[i,j,::]  = rot2d180(a[i,j,::] )
        return a

    class tconv(torch.nn.Module):
        def create_convTranspose2d(self):
            layers = nn.ModuleList()
            C2D = nn.ConvTranspose2d(
                n, m, kernel_size=k, stride=stride, dilation=1, bias=False, output_padding=0, groups=1)
            C2D.weight.data = torch.tensor(W, requires_grad=False)
            layers.append(C2D)
            return torch.nn.Sequential(*layers)

        def __init__(self):
            super(tconv, self).__init__()
            self.layers = self.create_convTranspose2d()

        def forward(self, x):
            return self.layers(x)



    class tconvWithConv(torch.nn.Module):
        def create_conv2d(self):
            layers = nn.ModuleList()
            C2D = nn.Conv2d(
                n, m, kernel_size=k, stride=1, dilation=1, bias=False)
            W_ = transposeConv2dKernel(W)
            C2D.weight.data = torch.tensor(W_, requires_grad=False)
            layers.append(C2D)
            return torch.nn.Sequential(*layers)

        def __init__(self):
            super(tconvWithConv, self).__init__()
            self.layers = self.create_conv2d()

        def forward(self, x):
            return self.layers(x)


    # input
    input = np.random.randn(N, n, h, w) * 1

    # stride, padd input
    hi = (h - 1) * stride + 1 + 2 * (k - 1)
    wi = (w - 1) * stride + 1 + 2 * (k - 1)
    inputStridedPadded = np.zeros(shape=(N, n, hi, wi))
    rh = np.arange(start=k - 1, step=stride, stop=(hi) - (k - 1))
    rw = np.arange(start=k - 1, step=stride, stop=(wi) - (k - 1))
    xv, yv = np.meshgrid(rh, rw)
    inputStridedPadded[:, :, yv, xv] = input[:, :, ::]

    oTconv =  tconv().forward(torch.tensor(input).float()).detach().numpy()
    oTconvWconv = tconvWithConv().forward(torch.tensor(inputStridedPadded).float()).detach().numpy()

```

cc @albanD @mruberry @jbschlosser

**Comment by Dimitri1** [comment]
I think that I have my answer.What you call **ConvTranspose2d** is actually a generalisation of mathematical deconvolution (1d), in multi channel 4d nxmxhxw  way.

So why don't call it "deconvolution 2d" or something? ConvTranspose2s is really confusing.

**Comment by jbschlosser** [comment]
It isn't named "deconvolution" because it isn't a true deconvolution AKA a true inverse of convolution. The answer [here](https://datascience.stackexchange.com/a/12110) should hopefully be helpful. Note that [Keras](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2DTranspose) follows a similar naming scheme as well.

Docs could be updated to reference these [convolution visualizations](https://github.com/vdumoulin/conv_arithmetic) and the [Deconvolutional Networks](https://www.matthewzeiler.com/mattzeiler/deconvolutionalnetworks.pdf) paper for more info.
